src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperContext.cs

```diff
@@ -18,15 +18,19 @@ public class TagHelperContext
         /// Instantiates a new <see cref="TagHelperContext"/>.
         /// </summary>
         /// <param name="allAttributes">Every attribute associated with the current HTML element.</param>
+        /// <param name="items">TODO</param>
         /// <param name="uniqueId">The unique identifier for the source element this <see cref="TagHelperContext" /> 
         /// applies to.</param>
         /// <param name="getChildContentAsync">A delegate used to execute and retrieve the rendered child content 
         /// asynchronously.</param>
-        public TagHelperContext([NotNull] IDictionary<string, object> allAttributes,
-                                [NotNull] string uniqueId,
-                                [NotNull] Func<Task<string>> getChildContentAsync)
+        public TagHelperContext(
+            [NotNull] IDictionary<string, object> allAttributes,
+            [NotNull] IDictionary<string, object> items,
+            [NotNull] string uniqueId,
+            [NotNull] Func<Task<string>> getChildContentAsync)
         {
```

Comment:
Can definitely provide two constructors. However don't want to have the `TagHelperContext` take in the `TagHelperContext` that was used to run the parent set of `TagHelper`s (that's more the `TagHelperExecutionContext`'s responsibility).

The `TagHelperContext` is currently more like the `TagHelperRunningContext` because it only exists within the `TagHelperRunner`. It was designed this way so users could potentially replace just the `TagHelperRunner` and build custom `TagHelper`s (aka call into their own versions `Process` methods with different parameters).